src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1108,6 +1109,51 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
       throw failures.failure(info, elementsShouldSatisfy(actual, unsatisfiedRequirements, info));
   }
 
+  /**
+   * Verifies that the all the given consumers can be satisfied by elements in the iterable with
+   * an element at most satisfies one consumer. No order requirement.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code Iterable}.
+   * @param consumers the consumers that are expected to be satisfied by the elements of the given {@code Iterable}.
+   * @throws NullPointerException if the given consumers is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if any {@code Consumer} in the consumers cannot be satisfied by elements in the given {@code Iterable}.
+   */
+  @SafeVarargs
+  public final <E> void assertSatisfy(AssertionInfo info, Iterable<? extends E> actual, Consumer<? super E>... consumers) {
+    assertNotNull(info, actual);
+    requireNonNull(consumers, "The Consumer<? super E>... expressing the assertions consumers must not be null");
+    requireNonNull(consumers[0], "The Consumer<? super E>... expressing the assertions consumers must not be null");
+    List<E>[] stasfiedElementsLists = new ArrayList[consumers.length];
+    for (int i = 0; i < consumers.length; i++) {
+      Consumer<? super E> consumer = consumers[i];
+      stasfiedElementsLists[i] = stream(actual).filter(element -> {
+        try {
+          consumer.accept(element);
+        } catch (AssertionError ex) {
+          return false;
+        }
+        return true;
+      }).collect(toList());
+    }
+    if (!isSatisfied(stasfiedElementsLists, 0))
+      throw failures.failure(info, shouldSatisfy(actual));
+  }
+
+  private static <E> boolean isSatisfied(List<E>[] stasfiedElementsLists, int begin) {
+    if (begin == stasfiedElementsLists.length) return true;
+    if (stasfiedElementsLists[begin].size() == 0) return false;
+
+    for (E element : stasfiedElementsLists[begin]) {
+      List<E>[] listsCopy = stasfiedElementsLists.clone();
+      for (int i = begin + 1; i < stasfiedElementsLists.length; i++)
```

Comment:
Typo in variable name "stasfiedElementsList".
```suggestion
  private static <E> boolean isSatisfied(List<E>[] satisfiedElementsLists, int begin) {
    if (begin == satisfiedElementsLists.length) return true;
    if (satisfiedElementsLists[begin].size() == 0) return false;

    for (E element : satisfiedElementsLists[begin]) {
      List<E>[] listsCopy = satisfiedElementsLists.clone();
      for (int i = begin + 1; i < satisfiedElementsLists.length; i++)
```